src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/IssuanceServerClient.kt

```diff
@@ -166,18 +165,14 @@ internal class IssuanceServerClient(
         request: CredentialIssuanceRequest.SingleRequest,
     ): CredentialIssuanceResponse =
         if (response.status.isSuccess()) {
-            when (issuerMetadata.credentialResponseEncryption) {
-                is CredentialResponseEncryption.NotRequired -> {
+            when (val encryptionSpec = request.encryption) {
+                null -> {
                     val success = response.body<SingleIssuanceSuccessResponse>()
                     success.toDomain()
                 }
 
-                is CredentialResponseEncryption.Required -> {
+                else -> {
```

Comment:
Why use else? 

There are two distinct cases: 
- Required (which corresponds to the code of else)
- SupportedNotRequired 
The later should be governed by a configuration option, IMO